packages/pwa/app/utils/routes-utils.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getUrlConfig} from './utils'
+import {urlParamTypes} from '../constants'
+
+/**
+ *  a function returns basePath string for locale/siteId/siteAlias
+ *  by reading over the url config inside the pwa-kit-config.json file
+ * @param locale
+ * @param siteId
+ * @returns {string}
+ */
+export const getLocaleAndSiteBasePath = (locale, siteId) => {
```

Comment:
A couple comments on this function:

1. I think the name is a little too specific. What happens if we expand on this function, will we have to change the name. I noticed that you use this utility like `const basepath = getLocaleAndSiteBasePath(locale)`. Why not call it simply `getBasePath()`
2. The arguments again are very specific, what would happen if we now wanted to add the ability to configu the url with the currency? You'd start to have a endless number of arguments. Might look better like `getBasePath(configValues)` or something similar.
3. For `localeReplacement` and `siteIdReplacement`, _replacement_ is the wrong word. I think you meant to use _placement_.
4. You have a condition ensuring that values for locale and siteId are actually passed in before pushing them onto the paths array. Would it be an exception is the config said to place the locale in the PATH, but we didn't supply a value?

Reply:
Thanks for the comment, `getBastPath` makes sense to me. 👍